src/main/kotlin/ecommerce/service/OrderService.kt

```diff
@@ -0,0 +1,125 @@
+        member: Member,
+        req: OrderRequest,
+    ): OrderResponse {
+        val product =
+            productRepository.findByIdOrNull(req.productId)
+                ?: throw NotFoundException("Product not found")
+
+        val option =
+            optionRepository.findByIdOrNull(req.optionId)
+                ?: throw NotFoundException("Option not found")
+
+        if (option.product?.id != product.id) {
+            throw IllegalArgumentException("Selected option does not belong to the specified product.")
+        }
+
+        if (option.quantity < req.quantity) {
+            throw IllegalArgumentException("Not enough stock available")
+        }
+
+        val totalAmount = calculateAmount(product.price.toInt(), req.quantity)
+        val paymentRequest =
+            PaymentRequest(
+                amount = totalAmount,
+                currency = req.currency,
+                paymentMethod = req.paymentMethod,
+            )
+
+        val paymentResponse = stripeClient.createCheckoutSession(paymentRequest)
+
+        if (paymentResponse == null || paymentResponse.id.isBlank()) {
+            throw StripeClientException.from(paymentResponse?.errorMessage ?: "Stripe did not return a valid response.")
+        }
+
+        option.quantity -= req.quantity
+        optionRepository.save(option)
+
+        val existingCartItem = cartItemRepository.findByMemberAndProduct(member, product)
+        if (existingCartItem.isPresent) {
+            cartItemRepository.delete(existingCartItem.get())
+        }
+
+        val order =
+            Order(
+                member = member,
+                status = OrderStatus.PAID,
+                createdAt = LocalDateTime.now(),
+            )
+
+        val orderItem =
+            OrderItem(
+                product = product,
+                option = option,
+                quantity = req.quantity,
+            )
+        order.addItem(orderItem)
+
+        val payment =
+            Payment(
+                order = order,
+                paymentIntentId = paymentResponse.id,
+                amount = totalAmount,
+                currency = req.currency,
+            )
+        order.assignPayment(payment)
+
+        orderRepository.save(order)
+
+        return OrderResponse(
+            success = true,
+            message = "Order placed successfully",
+            paymentIntentId = paymentResponse.id,
+        )
+    }
+
+    @Transactional
+    fun getOrdersForMember(member: Member): List<OrderResponse> {
+        val orders = orderRepository.findAllByMember(member)
+        return orders.map {
+            OrderResponse(
+                success = true,
+                message = "Order with ${it.items.size} item(s)",
+                paymentIntentId = it.payment?.paymentIntentId ?: "N/A",
+            )
+        }
+    }
+
+    private fun calculateAmount(
+        price: Int,
+        quantity: Int,
+    ): Int {
+        return price * quantity
+    }
```

Comment:
The `price` in `Product` is a `Double`, so forcibly converting it to `Int` for calculations seems odd. 🤔